calico_cni_k8s_test.go

```diff
@@ -1245,7 +1253,85 @@ var _ = Describe("CalicoCni", func() {
 					Expect(endpoints.Items[0].Spec.Endpoint).Should(Equal("eth0"))
 					Expect(endpoints.Items[0].Spec.ContainerID).Should(Equal(containerID))
 
-					_, err = testutils.DeleteContainer(netconf, contNs.Path(), name, testutils.K8S_TEST_NS)
+					// Now we create another pod with a very similar name.
```

Comment:
I think you should be calling out what you mean by "very similar".  Again - if you were reading this code for the first time would you get the subtleties of what's being tested here?